api/src/main/java/org/pmiops/workbench/workspaces/resources/WorkspaceResourceMapper.java

```diff
@@ -193,122 +143,55 @@ default WorkspaceResource fromDbUserRecentResource(
+  default ResourceFields fromDbUserRecentlyModifiedResource(
       DbUserRecentlyModifiedResource dbUserRecentlyModifiedResource,
-      FirecloudWorkspaceResponse fcWorkspace,
-      DbWorkspace dbWorkspace,
       CohortService cohortService,
       CohortReviewService cohortReviewService,
       ConceptSetService conceptSetService,
       DataSetService dataSetService) {
+
+    // null if Notebook, Long id otherwise
+    final Long resourceId = Longs.tryParse(dbUserRecentlyModifiedResource.getResourceId());
     final long workspaceId = dbUserRecentlyModifiedResource.getWorkspaceId();
+
     switch (dbUserRecentlyModifiedResource.getResourceType()) {
       case COHORT:
-        return fromDbUserRecentlyModifiedResourceAndCohort(
-            dbUserRecentlyModifiedResource,
-            fcWorkspace,
-            dbWorkspace,
-            cohortService.findDbCohortByCohortId(
-                getResourceIdInLong(dbUserRecentlyModifiedResource)));
+        return fromDbCohort(cohortService.findDbCohortByCohortId(resourceId));
+      case COHORT_REVIEW:
+        return fromCohortReview(
+            cohortReviewService.findCohortReviewForWorkspace(workspaceId, resourceId));
       case CONCEPT_SET:
-        return fromDbUserRecentlyModifiedResourceAndConceptSet(
-            dbUserRecentlyModifiedResource,
-            fcWorkspace,
-            dbWorkspace,
-            conceptSetService.getDbConceptSet(
-                workspaceId, getResourceIdInLong(dbUserRecentlyModifiedResource)));
+        return fromDbConceptSet(conceptSetService.getDbConceptSet(workspaceId, resourceId));
       case DATA_SET:
-        return fromDbUserRecentlyModifiedResourceAndDataSet(
-            dbUserRecentlyModifiedResource,
-            fcWorkspace,
-            dbWorkspace,
-            dataSetService
-                .getDbDataSet(workspaceId, getResourceIdInLong(dbUserRecentlyModifiedResource))
-                .orElseThrow(
-                    () ->
-                        new NotFoundException(
-                            String.format(
-                                "Dataset not found for dataSetId: %s",
-                                dbUserRecentlyModifiedResource.getResourceId()))));
+        return fromDbDataset(dataSetService.mustGetDbDataset(workspaceId, resourceId));
       case NOTEBOOK:
-        return fromDbUserRecentlyModifiedResourceAndNotebookName(
-            dbUserRecentlyModifiedResource,
-            fcWorkspace,
-            dbWorkspace,
-            dbUserRecentlyModifiedResource.getResourceId());
-      case COHORT_REVIEW:
-        return fromDbUserRecentlyModifiedResourceAndCohortReview(
-            dbUserRecentlyModifiedResource,
-            fcWorkspace,
-            dbWorkspace,
-            cohortReviewService.findCohortReviewForWorkspace(
-                workspaceId, getResourceIdInLong(dbUserRecentlyModifiedResource)));
-
+        return fromNotebookNameAndLastModified(
```

Comment:
It looks like fromNotebookNameAndLastModified is replacing fromDbUserRecentlyModifiedResourceAndNotebookName here, but it looks like the former returns ResourceFields and the latter returns WorkspaceResource. Do the callers of this function need to be updated?